### DISCUSSION TITLE
Improve - Signup flow

### DISCUSSION THREAD

**Discussion by davehakkens** [ONEARMY/community-platform#3236] [Big Updates]
Signup is a crucial first step for people using the platform. It’s needed to download, contact, comment and upload content. But it's not optimal

The goal here is to identify the problems and discuss/share possible ways forward to see if it's worth putting on our roadmap.

# Current problems
1. Email confirmations don't work (also part of [notifications](https://github.com/ONEARMY/community-platform/discussions/3235))
2. Member profiles avatar upload is large and non square  ([issue](https://github.com/ONEARMY/community-platform/issues/1589))
3. Users with incomplete profiles can comment, higher risk to spam   or bad interactions
4. Profiles settings had many features added in the recent years.  UX/UI is a big clunky.

# Things to keep in mind
- We have two types of profiles. Groups (spaces) and individuals (Members). The aim is to keep member profiles basic and simple and focus on extending groups
- We aim to have complete user profiles, this make the experience much better when browsing or interacting with others, thats why we have a few layers of profile completion based on whats wanted. The below image gives an overview 
<img width="389" alt="afbeelding" src="https://github.com/ONEARMY/community-platform/assets/13672737/edc91a59-0893-4128-a961-3d18d7671165">

# Whats needed next
- [ ] Fix email confirmation flow (and style emails if possible)
- [ ] Add square/cropped avatar upload members
- [ ] Set the right roles for commenting and make sure this is clear to the user. (According to chart above)
- [ ] Redesign the profile settings page to make it a good user experience. (there is a [PR open](https://github.com/ONEARMY/community-platform/pull/3094) to improve map section)
- [ ] Make signup flow mobile ready

**Comment by davehakkens** [comment]
@benfurber is working on this. Mapping out the issues and what to improve
<img width="1101" alt="afbeelding" src="https://github.com/ONEARMY/community-platform/assets/13672737/46164527-55dd-4fd5-8e61-7eb69929c013">

**Comment by benfurber** [comment]
I'm thinking of setting up a board for this epic so that it's clear what's inflight, obvious must haves and what should be discussed further/might need more design.

**Reply by benfurber** [reply]
Setup here: https://github.com/orgs/ONEARMY/projects/10/views/1

Useful?

**Reply by m-azoubel** [reply]
Hey Ben, I can't see the board

**Reply by m-azoubel** [reply]
<img width="959" alt="image" src="https://github.com/ONEARMY/community-platform/assets/90778074/866c0031-b45d-440b-b824-c2e7eac9fb63">

**Comment by benfurber** [comment]
One challenge atm is getting visibility on why it appears that so many confirmation emails don't get through. It's easy to guess but no actual data. 

It looks like we won't be able to get more visibility on it with the default firebase provider and so changing to something else is essential.

https://github.com/ONEARMY/community-platform/issues/3250

**Comment by benfurber** [comment]
I think it's worth making the sign-up journey more exciting and on-brand. A quick idea is something like this. Would love a more talented illustrator to have a go!

<img width="638" alt="Screenshot 2024-02-12 at 10 25 26" src="https://github.com/ONEARMY/community-platform/assets/16688508/27f7d2e4-f795-42a1-92ab-8f9f6679940c">

<img width="856" alt="Screenshot 2024-02-12 at 10 24 55" src="https://github.com/ONEARMY/community-platform/assets/16688508/a06047ac-8e27-4f98-a6d2-7f9b998fec6a">

**Reply by davehakkens** [reply]
I like the idea of this a lot. But would suggest to wait with the UI of this until we dive a bit more deeper in design the upcoming months. To avoid having to redo it.

That said we can think of intergrating UX features and functions that later can be properly designed.
(For instance that "sign up to send a message" on the contact page for not logged in users currently doest exist but that function would make a lot of sense to intergrate

**Comment by benfurber** [comment]
I think we should make it really clear where in the sign-up journey a user is, close to something like this.

<img width="1157" alt="Screenshot 2024-02-12 at 10 27 04" src="https://github.com/ONEARMY/community-platform/assets/16688508/82ab1a42-c815-4bee-98b9-89466800c17e">

<img width="1157" alt="Screenshot 2024-02-12 at 10 27 37" src="https://github.com/ONEARMY/community-platform/assets/16688508/65426b68-81ef-472c-9543-288cd28b63b3">

**Comment by benfurber** [comment]
As well as 'blocking' a user from commenting/creating content on a technical level, we need some design work on how to present that information to users and giving them the easy action to complete, so that they can comment, etc.

**Comment by benfurber** [comment]
I think the cropping issue needs an update on what'd going on. I'm going to check the user flow.

**Reply by benfurber** [reply]
I think I'm just going to spike the feature this afternoon.

**Reply by benfurber** [reply]
Turns out it's all a bit messy in the image upload space! I think it needs a tidy-up: https://github.com/ONEARMY/community-platform/issues/3270
